roles/kafka_controller/tasks/get_meta_properties.yml

```diff
@@ -20,9 +20,25 @@
   register: zoo_cluster
   when: kraft_migration|bool
 
-- name: Format Data Directory
-  shell: "{{ binary_base_path }}/bin/kafka-storage  format -t={{ clusterid }} -c {{ kafka_controller.config_file }}  --ignore-formatted"
+- name: Create SCRAM Credentials
+  set_fact:
+    scram_credentials: >-
+      {% if 'SCRAM-SHA-512' in kafka_sasl_enabled_mechanisms %}
+      SCRAM-SHA-512=[name={{ sasl_scram_users_final.admin.principal }},password={{ sasl_scram_users_final.admin.password }}]
+      {% endif %}
+      {% if 'SCRAM-SHA-512' in kafka_sasl_enabled_mechanisms and 'SCRAM-SHA-256' in kafka_sasl_enabled_mechanisms %},{% endif %}
+      {% if 'SCRAM-SHA-256' in kafka_sasl_enabled_mechanisms %}
+      SCRAM-SHA-256=[name={{ sasl_scram256_users_final.admin.principal }},password={{ sasl_scram256_users_final.admin.password }}]
+      {% endif %}
+  when: "'SCRAM-SHA-512' in kafka_sasl_enabled_mechanisms or 'SCRAM-SHA-256' in kafka_sasl_enabled_mechanisms"
+
+- name: Format Data Directory and Add Scram Credentials
```

Comment:
Have you tested this command in case of upgrades / zk to kraft migration ? 
This often causes some trouble  in terms of file owner/groups during migration

Reply:
Tested and added link in PR description.